### ISSUE TITLE
How do I allow phone music to play while using SCRecorder?

### ISSUE THREAD

**Issue by mschettler** [rFlex/SCRecorder#127]
Whenever I enable my SCRecorder for video, it stops the background music from playing

 [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryAmbient error:nil];

does not work.

How can I keep the music playing like Snapchat now does?

**Comment by mschettler** [comment]
bump

**Comment by huntermonk** [comment]
I haven't been able to figure this out as well, any help is much appreciated.

**Comment by StormFactory** [comment]
I don't use Snapchat, so I'm not sure how they do it.  Are you looking to be able to record a video while music plays so you can record audio and the music at the same time?  For example like a karaoke app might do?

**Comment by maximillionnn** [comment]
Correct, that is the desired effect. More specifically, I want audio from other apps to be playing while I record.

For example, I could have a song playing from spotify in the background while I record.

**Comment by mschettler** [comment]
@StormFactory  yes, exactly! right now, there doesn't seem to be a way, and I might have to refactor out SCRecorder simply for this reason

**Comment by mschettler** [comment]
@StormFactory any idea? @maximillionnn  @huntermonk

**Comment by StormFactory** [comment]
@huntermonk I need something similar on a project I'm starting this week so if I come up with a solution I will report back.

**Comment by huntermonk** [comment]
@StormFactory @mschettler Great, let me know if there's any way I can help. 

(deleted duplicate acc with username maximillionnn)

**Comment by huntermonk** [comment]
Hey @StormFactory , did you figure anything out here?

**Comment by MartinW** [comment]
@StormFactory @mschettler @huntermonk Has anyone made progress on this? Thanks

**Comment by huntermonk** [comment]
No progress yet. I've also noticed that if you get a phone call while the
recorder is open or a player is open, you get some odd behavior. Possibly
related.

**Comment by sumitsharma** [comment]
I got some hint from Apple Documentation and Stackoverflow. If you try this then background music will start playing irrespective of your application is coming into fore background or going into background.

https://developer.apple.com/library/ios/documentation/Audio/Conceptual/AudioSessionProgrammingGuide/ConfiguringanAudioSession/ConfiguringanAudioSession.html

http://stackoverflow.com/questions/28715597/allows-music-playback-during-recording-video-like-snapchat-ios/32416013#32416013

Now, the background music get pause for 1 sec and then continue onwards. This happens when you app come from background to foreground and vice versa.
